### DISCUSSION TITLE
Insights Global Variables - Add relation interfaces

### DISCUSSION THREAD

**Discussion by claytongulick** [directus/directus#14194] [Feature Requests]
I love the new global variable panels, they'll get us a far way towards a pretty robust BI solution inside of directus!

An additional feature that would bring this closer to Superset and other BI solutions would be to add the relation interfaces to the list of interface options.

A concrete use case: I want to run a health scorecard report for a patient. I need to be able to search for and select that patient the same way I would with a O2M field.

Bonus points for optional multi-select, like M2M? 😄

**Comment by benhaynes** [comment]
This sounds like a great capability... maybe I'm confused, but how would this differ from the Dynamic List panel? 🤔

**Reply by rijkvanzanten** [reply]
@benhaynes The OP is the same as "select another item" panel idea we were discussing before on discord 🙂

**Comment by azrikahar** [comment]
This has been sponsored and implemented in #15452 (released in `9.18.0`) 👍

**Reply by claytongulick** [reply]
I'm working on testing this out, but not having luck. I added a "patient" relational global variable, and was able to select one or multiple patients for it.

But when I try to add a filter that uses it with {{patient}} the filter doesn't save the {{patient}} part.

Even doing a simple query with a single measure is giving me a graphql error, I've tried both "_eq" and "_in":
```
{"errors":[{"message":"GraphQL validation error.","extensions":{"code":"GRAPHQL_VALIDATION_EXCEPTION","graphqlErrors":[{"message":"Field \"_eq\" is not defined by type \"patient_filter\". Did you mean \"_or\" or \"sex\"?","locations":[{"line":1,"column":116}]}],"stack":"Error: GraphQL validation error.\n    at GraphQLService.execute (/Users/claytongulick/source/directus_fork/api/src/services/graphql/index.ts:141:10)\n    at /Users/claytongulick/source/directus_fork/api/src/controllers/graphql.ts:36:38\n    at /Users/claytongulick/source/directus_fork/api/src/utils/async-handler.ts:4:18\n    at Layer.handle [as handle_request] (/Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/layer.js:95:5)\n    at trim_prefix (/Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/index.js:328:13)\n    at /Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/index.js:286:9\n    at Function.process_params (/Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/index.js:346:12)\n    at next (/Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/index.js:280:10)\n    at /Users/claytongulick/source/directus_fork/api/src/middleware/graphql.ts:66:9\n    at /Users/claytongulick/source/directus_fork/api/src/utils/async-handler.ts:4:18\n    at Layer.handle [as handle_request] (/Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/layer.js:95:5)\n    at trim_prefix (/Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/index.js:328:13)\n    at /Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/index.js:286:9\n    at Function.process_params (/Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/index.js:346:12)\n    at next (/Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/index.js:280:10)\n    at Function.handle (/Users/claytongulick/source/directus_fork/node_modules/.pnpm/express@4.18.1/node_modules/express/lib/router/index.js:175:3)"}}]}
```


<img width="1440" alt="Screen Shot 2022-09-20 at 1 19 43 PM" src="https://user-images.githubusercontent.com/695284/191334431-5e617cf2-8f5f-4743-97bd-6a380b899f0a.png">

**Reply by azrikahar** [reply]
Can you try going into the raw editor to make sure it's saved as described here: https://github.com/directus/directus/issues/14198#issuecomment-1251939560?

**Reply by claytongulick** [reply]
Thanks @azrikahar, but that doesn't seem to be the problem I'm experiencing - the raw editor won't save my values, I only get:

```
{
	"_and": [
		{
			"patient_id": {
				"_in": [
					null
				]
			}
		}
	]
}
```

I change null to {{patient_id}} as many times as I want, but the value doesn't save.

**Reply by azrikahar** [reply]
Can you share a quick screen capture of how were you changing it? Here's my attempt which seems to work for me on `9.18.1`, is it different from your attempts?

https://user-images.githubusercontent.com/42867097/192715500-1188ed68-d792-46e6-80cb-f183c89dd253.mp4

---

Here's another attempt creating the dashboard from scratch and using `_in`:

https://user-images.githubusercontent.com/42867097/192718402-d7fba98f-6ae2-4b3f-92a3-3a139ef654e6.mp4

**Reply by claytongulick** [reply]
thanks @azrikahar ! Here's a video (this is 9.18.1):


https://user-images.githubusercontent.com/695284/194139156-81ec98d0-1ba3-4da3-a20a-a336200b7584.mov

**Reply by claytongulick** [reply]
One thing I notice when I compare our videos, is when I add the {{patients}} filter, I don't get any sort of syntax highlighting for it, but you do.

I've made sure that I'm on the current version and have done a full rebuild of the admin app (pnpm -r build) & hard refresh a couple times to make sure it's current.

**Reply by azrikahar** [reply]
> One thing I notice when I compare our videos, is when I add the {{patients}} filter, I don't get any sort of syntax highlighting for it, but you do.

Nice catch! Seems like that's the difference indeed. Totally understand how the Raw Editor can be interpreted as Edit Raw Value due to similar naming, but the Raw Editor in this case would be the `{}` button beside the input labels, instead of the Edit Raw Value you were using:

https://user-images.githubusercontent.com/42867097/194232749-5fdff6d8-42eb-4f35-8a16-c13c7102acf7.mp4

In my videos you can also see I'm toggling it which changes the input directly, instead of appearing as a modal like Edit Raw Value 👍
